### ISSUE TITLE
line-and-point legends

### ISSUE THREAD

**Issue by majiang** [altair-viz/altair#2193]
I need a Line Chart with Points plots and [there is an example](https://altair-viz.github.io/gallery/line_chart_with_points.html).  It states that `altair.Chart(data).mark_line(point=True)` will produce line chart with points.

Since I have multiple series to plot I specified `color='kind'`, `strokeDash='kind'` (or both).  I found that the legends are only circles.

* With `color='kind'`, the circles are filled with the colors.
* With `strokeDash='kind'`, the boundary of the circles are solid or dashed lines.

Without `point=True` I have lines and the effects are that lines are colored or dashed.  With both `point=True` and {`color='kind'`, `strokeDash='kind'`, or both} I would expect the legends to have both lines and circles, same appearance as plot.  I couldn't figure out how to achieve the desired behavior so that I am opening this issue.

**Comment by jakevdp** [comment]
If you need more control over the exact look of points, I'd recommend using ``mark_point()``

**Comment by joelostblom** [comment]
@majiang  I am going through Altair issues to find those that have been resolved and can be closed. Would you be able to close this issue or add a comment if there is something you don't think is resolved yet?

**Comment by majiang** [comment]
Thanks, I'll have a look.

**Comment by majiang** [comment]
Sorry for not being able to try and feedback that time.

I now think it is impossible due to vega-lite's limitation and describe it with examples.

The desired output was something like
![image](https://user-images.githubusercontent.com/1482349/159658019-6d769f40-de88-474b-854b-f443fd21ffa4.png)

which I created with Excel.  By "line-and-point legend" I meant line legend and point legend combined as they look.

With altair, when I opened this issue,

```python
def sample_data():
    return pandas.DataFrame({
        "x": [0, 1, 2, 3, 0, 1, 2, 3],
        "y": [0, 1, 2, 3, 3, 2, 1, 0],
        "kind": ["a", "a", "a", "a", "b", "b", "b", "b"],
    })

altair.Chart(sample_data()).mark_line(point=True).encode(
    x="x", y="y", color="kind", strokeDash="kind")
```
produced

![visualization (3)](https://user-images.githubusercontent.com/1482349/159655708-e4ef8933-7858-4363-8358-b75c675cd5be.svg)

the main part of which was satisfactory, but only legend was not.

I have also tried
```python
(
    altair.Chart(sample_data()).mark_point().encode(
        x="x", y="y", color="kind"
    )
    +
    altair.Chart(sample_data()).mark_line().encode(
        x="x", y="y", color="kind", strokeDash="kind"
    )
)
```
which produces similar result:

![visualization (4)](https://user-images.githubusercontent.com/1482349/159657347-a78a092c-c3a3-486e-806a-a5bcc86a189d.svg)

Here I don't mind whether the circles are filled or not.

Since [the appearance of legend of a `Chart` is determined fully automatically by vega-lite from the definition of encoding channels](https://vega.github.io/vega-lite/docs/legend.html), we can't produce the desired output.
